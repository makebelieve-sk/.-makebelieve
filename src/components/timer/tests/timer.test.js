import React from 'react';
import renderer from 'react-test-renderer';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

import Timer from '../timer';

Enzyme.configure({ adapter: new Adapter() });

const setUp = (prop1, prop2) => shallow(
    <Timer getWinTime={prop1} timeEnd={prop2} />
);

const mock = {
    getWinTime: jest.fn(),
    timeEnd: jest.fn()
};

describe(`Timer component`, () => {
    let component;

    beforeEach(() => {
        component = setUp(mock.getWinTime, mock.timeEnd);
    });

    it(`should contain div wrapper`, () => {
        const wrapper = component.find('.timer_value');
        expect(wrapper.length).toBe(1);
    });

    it(`Timer correctly renders after relaunch`, () => {
        const tree = renderer  
            .create(<Timer 
                getWinTime={mock.getWinTime}
                timeEnd={mock.timeEnd}
            />)
            .toJSON();
    
        expect(tree).toMatchSnapshot();
    });
});

