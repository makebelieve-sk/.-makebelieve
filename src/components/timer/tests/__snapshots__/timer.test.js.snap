// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Timer component Timer correctly renders after relaunch 1`] = `
<div
  className="timer_value"
>
  <span
    className="timer_mins"
  >
    05
  </span>
  <span
    className="timer_dots"
  >
    :
  </span>
  <span
    className="timer_secs"
  >
    00
  </span>
</div>
`;
